src/core/agent/modules/signify/signifyApi.ts

```diff
@@ -105,6 +105,39 @@ export class SignifyApi {
     }
   }
 
+  async createDelegationIdentifier(delegatorPrefix: string): Promise<string> {
+    const signifyName = utils.uuid();
+    const icpResult2 = await this.signifyClient
+      .identifiers()
+      .create(signifyName, { delpre: delegatorPrefix });
+    const operation = await icpResult2.op();
+    // no need to wait for operation to complete because it needs to be interacted from delegator
+    const delegatePrefix = operation.name.split(".")[1];
+    return delegatePrefix;
+  }
+
+  async interactDelegation(signifyName: string, delegatePrefix: string) {
+    const anchor = {
+      i: delegatePrefix,
+      s: "0",
+      d: delegatePrefix,
+    };
+    return this.signifyClient.identifiers().interact(signifyName, anchor);
+  }
+
+  async checkDelegationSuccess(signifyName: string): Promise<boolean> {
```

Comment:
I presume this checks that the delegation is "complete"? As in the other wallet has added the anchor/seal into their KEL (`interactDelegation`)

Reply:
Correct

Reply:
OK, can we name it `delegationApproved`?